### PR TITLE
add option to remove partfile 1.1

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

**Comment by arvidn** [comment]
this needs a test

**Comment by sledgehammer999** [comment]
I tried this twice. The partfile isn't removed. Am I supposed to try this on top of #508?

**Comment by sledgehammer999** [comment]
`delete_files` doesn't remove the partfile either.

**Comment by arvidn** [comment]
yes, sorry. I kind of forgot about that other pull request. I believe on windows that one may be necessary for part file deletion to work reliably. It's possible that stopping the torrent first, and then deleting would get it to work.

My theory is that the file isn't closed, and you can't delete open files on windows. stopping the torrent should close the file handle to the partfile. And this is basically what that other PR does or attempts to do.

**Comment by arvidn** [comment]
thanks for testing btw. I'll put together a regression test for this as well. That hopefully can confirm the failure on windows as well.

**Comment by sledgehammer999** [comment]
I'll try to manually make the code change, because the 2 branches don't merge cleanly.
I'll report back soon.

File: src/storage.cpp [hunk 1]
```diff
@@ -779,50 +780,61 @@ namespace libtorrent
...
+				, end(directories.rend()); i != end; ++i)
+			{
+				error_code error;
+				delete_one_file(*i, error);
+				if (error && !ec) { ec.file = -1; ec.ec = error; ec.operation = storage_error::remove; }
+			}
+		}
+
+		if (options == session::delete_files
+			|| options == session::delete_partfile)
```
Comment (sledgehammer999) [inline src/storage.cpp]: I put the code from #508 inside this if.

**Comment by sledgehammer999** [comment]
I pointed to you where I put the #508 code.
Now delete_partfile deletes the partfile and leaves the files.
delete_files deletes both.

So I think you can merge both PRs.

**Comment by arvidn** [comment]
great, thanks! I'd like to get a regression test working before merging this one though

**Pull request merged**